src/ResourceManager/Insights/Commands.Insights/Diagnostics/SetAzureRmDiagnosticSettingCommand.cs

```diff
@@ -41,6 +41,27 @@ public class SetAzureRmDiagnosticSettingCommand : ManagementCmdletBase
         public string ResourceId { get; set; }
 
         /// <summary>
+        /// Gets or sets the switch for storage parameter of the cmdlet
+        /// </summary>
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "Disable storage")]
+        [ValidateNotNullOrEmpty]
+        public SwitchParameter DisableStorage { get; set; }
+
+        /// <summary>
+        /// Gets or sets the switch for eventhub parameter of the cmdlet
+        /// </summary>
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "Disable service bus")]
+        [ValidateNotNullOrEmpty]
+        public SwitchParameter DisableServiceBus { get; set; }
+
+        /// <summary>
+        /// Gets or sets the switch for workspace parameter of the cmdlet
+        /// </summary>
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "Disable workspace")]
+        [ValidateNotNullOrEmpty]
+        public SwitchParameter DisableWorkspace { get; set; }
```

Comment:
@vcanaa how do you handle the case of the three Disable parameters being used with the `Enabled` parameter? If a user shouldn't be able to use all four together (or `Enabled` with any combination of the three Disable parameters), then you should consider creating two parameters - one that has `Enabled`, and another that has the three Disable parameters